File: lib/domain/models/support_note.dart
class SupportNote {
  final String id;
  final String ticketId;
  final String assetId;
  final String handledBy;
  final String content;
  final DateTime createdAt;

  const SupportNote({
    required this.id,
    required this.ticketId,
    required this.assetId,
    required this.handledBy,
    required this.content,
    required this.createdAt,
  });

  factory SupportNote.fromJson(Map<String, dynamic> json) {
    return SupportNote(
      id: json['id'] as String? ?? '',
      ticketId: json['ticketId'] as String? ?? '',
      assetId: json['assetId'] as String? ?? '',
      handledBy: json['handledBy'] as String? ?? '',
      content: json['content'] as String? ?? '',
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt'] as String) : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'ticketId': ticketId,
      'assetId': assetId,
      'handledBy': handledBy,
      'content': content,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: backend/Entities/SupportNote.cs
namespace HoldMaintenance.Api.Entities;

public class SupportNote
{
    public string Id { get; set; } = string.Empty;
    public string TicketId { get; set; } = string.Empty;
    public string AssetId { get; set; } = string.Empty;
    public string HandledBy { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}
